app/controllers/admin/tenants_controller.rb

```diff
@@ -32,4 +33,13 @@ def update
     def tenant_params
       params.require(:tenant).permit(:name, :schema, :schema_type)
     end
+
+    def update_default_tenant_administrator
+      Tenant.switch(@tenant.schema) do
+        default_admin_account = User.administrators.first
+        default_admin_account.update!(email: current_user.email,
+                                      username: current_user.username,
+                                      encrypted_password: current_user.encrypted_password)
+      end
```

Comment:
This is pure conjecture since I haven't reproduced it.

I was jut wondering whether we could have an issue under certain conditions because `current_user` gets the user from the default schema but we're inside a `Tenant.switch` block :thinking:. It might be 100% safe because at this point `current_user` is probably cached, but, in any case, perhaps we could reduce the uncertainty with something like:

```suggestion
      login_params = {
        email: current_user.email,
        username: current_user.username,
        encrypted_password: current_user.encrypted_password
      }

      Tenant.switch(@tenant.schema) do
        default_admin_account = User.administrators.first
        default_admin_account.update!(login_params)
      end
```

Do you think it's worth it? Or am I being paranoid?

Reply:
That's good paranoid. I also realized this during development. However, as the current user returned the correct user, I decided to dismiss it. Anyway, it would be better to be sure by doing what you suggest.